pkg/miniogw/gateway-storj.go

```diff
@@ -200,38 +200,46 @@ func (s *storjObjects) ListBuckets(ctx context.Context) (
 func (s *storjObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
 	delimiter string, maxKeys int) (result minio.ListObjectsInfo, err error) {
 	defer mon.Task()(&ctx)(&err)
+
+	if delimiter != "" && delimiter != "/" {
+		return minio.ListObjectsInfo{}, Error.New("delimiter %s not supported", delimiter)
+	}
+
 	startAfter := paths.New(marker)
-	var fl []minio.ObjectInfo
+	var objects []minio.ObjectInfo
+	var prefixes []string
 	o, err := s.storj.bs.GetObjectStore(ctx, bucket)
 	if err != nil {
 		return minio.ListObjectsInfo{}, err
 	}
-	items, more, err := o.List(ctx, paths.New(prefix), startAfter, nil, true, maxKeys, meta.All)
+	items, more, err := o.List(ctx, paths.New(prefix), startAfter, nil, delimiter == "", maxKeys, meta.All)
 	if err != nil {
 		return result, err
 	}
 	if len(items) > 0 {
-		//Populate the objectlist (aka filelist)
-		f := make([]minio.ObjectInfo, len(items))
-		for i, fi := range items {
-			f[i] = minio.ObjectInfo{
-				Bucket:      bucket,
-				IsDir:       fi.IsPrefix,
-				Name:        fi.Path.String(),
-				ModTime:     fi.Meta.Modified,
-				Size:        fi.Meta.Size,
-				ContentType: fi.Meta.ContentType,
-				UserDefined: fi.Meta.UserDefined,
-				ETag:        fi.Meta.Checksum,
+		for _, item := range items {
+			if item.IsPrefix {
+				prefixes = append(prefixes, item.Path.String()+"/")
```

Comment:
use a continue here and unindent the else branch (https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow), it keeps the normal flow to the left

Reply:
done